### PR TITLE
Removed duplication of ExtBufWrapper class

### PR DESCRIPTION
**Pull request by dmitryermilov**
Signed-off-by: Dmitry Ermilov <dmitry.ermilov@intel.com>

### CODE REVIEW

File: samples/sample_common/include/sample_utils.h [hunk 1]
```diff
@@ -625,6 +696,207 @@ void init_ext_buffer(T & ext_buffer)
...
+    { };
+
+    bool operator () (mfxExtBuffer* b)
+    {
+        return  (b && b->BufferId == id);
+    };
+};
+
+template<typename T>
+class ExtBufWrapper: public T
```
Comment (mgonchar) [inline samples/sample_common/include/sample_utils.h]: Name is misleading, I believe it should be something like:
```
ExtBufManagementWrapper
```

File: samples/sample_common/include/sample_utils.h [hunk 2]
```diff
@@ -625,6 +696,207 @@ void init_ext_buffer(T & ext_buffer)
...
+            if (buf_id == allowed[i])
+                return true;
+        }
+
+        return false;
+    }
+
+    typedef std::vector<mfxExtBuffer*>::iterator ExtBufIterator;
+    typedef std::vector<mfxExtBuffer*>::const_iterator constExtBufIterator;
+    std::vector<mfxExtBuffer*> m_ext_buf;
```
Comment (mgonchar) [inline samples/sample_common/include/sample_utils.h]: I think we should avoid all low-level memory operations as much as possible, so we could have some other member for storage:
```
std::vector<std::unique_ptr<mfxU8[]>> m_memory_storage;
```

File: samples/sample_common/include/sample_utils.h [hunk 3]
```diff
@@ -625,6 +696,207 @@ void init_ext_buffer(T & ext_buffer)
...
+
+    template <typename TB>
+    TB* GetExtBuffer()
+    {
+        ExtBufIterator it = std::find_if(m_ext_buf.begin(), m_ext_buf.end(), CmpExtBufById(mfx_ext_buffer_id<TB>::id));
+        if (it != m_ext_buf.end())
+        {
+            return (TB*)*it;
+        }
+        return 0;
```
Comment (mgonchar) [inline samples/sample_common/include/sample_utils.h]: let's replace all ```0``` to ```nullptr```

File: samples/sample_common/include/sample_utils.h [hunk 4]
```diff
@@ -625,6 +696,207 @@ void init_ext_buffer(T & ext_buffer)
...
+    ExtBufWrapper(const ExtBufWrapper& ref)
+    {
+        *this = ref; // call to operator=
+    }
+
+    ExtBufWrapper& operator=(const ExtBufWrapper& ref)
+    {
+        const T* src_base = &ref;
+        return operator=(*src_base);
+    }
```
Comment (mgonchar) [inline samples/sample_common/include/sample_utils.h]: Additionaly move ctor and move assignment should be defaulted

File: samples/sample_common/include/sample_utils.h [hunk 5]
```diff
@@ -625,6 +696,207 @@ void init_ext_buffer(T & ext_buffer)
...
+
+    ExtBufWrapper& operator=(const T& ref)
+    {
+        // copy content of main structure type T
+        T* dst_base = this;
+        const T* src_base = &ref;
+        Copy(*dst_base, *src_base);
+
+        //remove all existing extended buffers
```
Comment (mgonchar) [inline samples/sample_common/include/sample_utils.h]: extension buffers :)

File: samples/sample_common/include/sample_utils.h [hunk 6]
```diff
@@ -625,6 +696,207 @@ void init_ext_buffer(T & ext_buffer)
...
+    };
+};
+
+template<typename T>
+class ExtBufWrapper: public T
+{
+public:
+    ExtBufWrapper()
+    {
+        T& base = *this;
```
[the remark this answers is not in the document]
Comment (dmitryermilov) [reply samples/sample_common/include/sample_utils.h]: Ok. Thanks!

**Pull request merged**